src/monitoring_service/handlers.py

```diff
@@ -406,67 +408,84 @@ def action_monitoring_triggered_event_handler(event: Event, context: Context) ->
     if channel.update_status:
         last_onchain_nonce = channel.update_status.nonce
 
-    user_address = monitor_request.non_closing_signer
-    user_deposit = context.user_deposit_contract.functions.effectiveBalance(user_address).call()
-
     if monitor_request.nonce <= last_onchain_nonce:
         log.info(
             "Another MS submitted the last known channel state", monitor_request=monitor_request
         )
+        return
 
+    latest_block = context.w3.eth.blockNumber
+    last_confirmed_block = latest_block - context.required_confirmations
```

Comment:
Mh, we also have `context.last_known_block`. I guess one of them should go

Reply:
I intentionally did not use `context.last_known_block`, because the initial sync will do huge amounts of calls when you do a call for each block between `last_known_block` and current block (which might be a lot) in these cases.
Also using the last confirmed block makes the logic more clear, since we do not care about which blocks have already been processed for this.

Reply:
Nice that you thought about that already. Let's leave it like that then.

But do you know why we have `last_known_block` duplicated?

Reply:
Probably because we forget that `last_known_block` is available from `context.ms_state.blockchain_state.latest_known_block` at some points, so we added it to the context? I'll make add a `last_known_block` property to `Context` that return context.ms_state.blockchain_state.latest_known_block`, so that we can easily access it and won't add it again in the same manner sometime in the future.